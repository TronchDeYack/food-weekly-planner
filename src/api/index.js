const recipeFake = {
  id: 'SPAGHETTI-ID',
  title: 'Spaghetti bolognaise',
  minutes: 20,
  caloriesLevel: 'medium',
  ingredients: [
    {
      id: 'PATE-ID',
      label: 'Pâtes',
      quantity: 200,
      unit: 'GR',
      type: 'STARCHY-FOOD',
    },
    {
      id: 'SAUCE-TOMATE-ID',
      label: 'Sauce Tomate',
      quantity: 200,
      unit: 'GR',
      type: 'CONDIMENT',
    },
    {
      id: 'STEAK-ID',
      label: 'Steak haché',
      quantity: 100,
      unit: 'GR',
      type: 'MEAT',
    },
    {
      id: 'OIL-ID',
      label: 'Huile d\'olives',
      quantity: 3,
      unit: 'CAC',
      type: 'CONDIMENT',
    },
  ],
};

const recipesFake = [
  recipeFake,
  {
    id: 'SOUPE-ID',
    title: 'Soupe à la tomate',
    minutes: 30,
    caloriesLevel: 'low',
    ingredients: [
      {
        id: 'TOMATE-ID',
        label: 'Tomates',
        quantity: 4,
        unit: 'UNIT',
        type: 'VEGETABLE',
      },
      {
        id: 'SALT-ID',
        label: 'Sel',
        quantity: 2,
        unit: 'CAC',
        type: 'CONDIMENT',
      },
    ],
  },
  {
    id: 'WELSH-ID',
    title: 'Welsh',
    minutes: 45,
    caloriesLevel: 'high',
    ingredients: [
      {
        id: 'CHEDDAR-ID',
        label: 'Cheddar',
        quantity: 400,
        unit: 'GR',
        type: 'DAIRY-PRODUCT',
      },
      {
        id: 'BEER-ID',
        label: 'Bière brune',
        quantity: 25,
        unit: 'CL',
        type: 'OTHER',
      },
      {
        id: 'HAM-ID',
        label: 'Jambon',
        quantity: 100,
        unit: 'GR',
        type: 'MEAT',
      },
      {
        id: 'BREAD-ID',
        label: 'Pain',
        quantity: 4,
        unit: 'SLICE',
        type: 'STARCHY-FOOD',
      },
    ],
  },
];

export function getRecipe(id) {
  console.log(`get recipe ${id}`);
  return Promise.resolve(recipeFake);
}

export function getRecipes() {
  return Promise.resolve(recipesFake);
}

export function getRecipesOfTheWeek() {
  return Promise.resolve([recipesFake[1], recipesFake[2]]);
}

export function getShoppingList() {
  return Promise.resolve([
    {
      id: 'BREAD-ID',
      label: 'Pain',
      quantity: 1,
      unit: 'unit',
      type: 'STARCHY-FOOD',
      bought: false,
    },
    {
      id: 'CHEDDAR-ID',
      label: 'Cheddar',
      quantity: 400,
      unit: 'GR',
      type: 'DAIRY-PRODUCT',
      bought: true,
    },
    {
      id: 'MILK-ID',
      label: 'Lait',
      quantity: 1,
      unit: 'L',
      type: 'DAIRY-PRODUCT',
      bought: false,
    },
    {
      id: 'BEER-ID',
      label: 'Bière brune',
      quantity: 25,
      unit: 'CL',
      type: 'OTHER',
      bought: false,
    },
    {
      id: 'HAM-ID',
      label: 'Jambon',
      quantity: 100,
      unit: 'GR',
      type: 'MEAT',
      bought: true,
    },
    {
      id: 'CHICKEN-ID',
      label: 'Piou piou',
      quantity: 200,
      unit: 'GR',
      type: 'MEAT',
      bought: false,
    },
  ]);
}

export function getIngredients() {
  return Promise.resolve([
    {
      id: 'PATE-ID',
      label: 'Pâtes',
      quantity: 200,
      unit: 'GR',
      type: 'STARCHY-FOOD',
    },
    {
      id: 'SAUCE-TOMATE-ID',
      label: 'Sauce Tomate',
      quantity: 200,
      unit: 'GR',
      type: 'CONDIMENT',
    },
    {
      id: 'STEAK-ID',
      label: 'Steak haché',
      quantity: 100,
      unit: 'GR',
      type: 'MEAT',
    },
    {
      id: 'OIL-ID',
      label: 'Huile d\'olives',
      quantity: 3,
      unit: 'CAC',
      type: 'CONDIMENT',
    },
    {
      id: 'TOMATE-ID',
      label: 'Tomates',
      quantity: 4,
      unit: 'UNIT',
      type: 'VEGETABLE',
    },
    {
      id: 'SALT-ID',
      label: 'Sel',
      quantity: 2,
      unit: 'CAC',
      type: 'CONDIMENT',
    },
    {
      id: 'CHEDDAR-ID',
      label: 'Cheddar',
      quantity: 400,
      unit: 'GR',
      type: 'DAIRY-PRODUCT',
    },
    {
      id: 'BEER-ID',
      label: 'Bière brune',
      quantity: 25,
      unit: 'CL',
      type: 'OTHER',
    },
    {
      id: 'HAM-ID',
      label: 'Jambon',
      quantity: 100,
      unit: 'GR',
      type: 'MEAT',
    },
    {
      id: 'BREAD-ID',
      label: 'Pain',
      quantity: 4,
      unit: 'SLICE',
      type: 'STARCHY-FOOD',
    },
  ]);
}
