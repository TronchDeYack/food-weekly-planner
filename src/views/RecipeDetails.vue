<template>
  <v-container>
    <v-row>
      <v-col cols=12>
        <h3>Spaghetti Bolognaise</h3>
      </v-col>
    </v-row>
    <v-row align="center">
      <v-col class="d-flex align-center">
        <Timer />
        20 min
      </v-col>
      <v-col>
        <Calories class="medium"/>
      </v-col>
    </v-row>
    <v-divider/>
    <v-row>
      <v-col cols=12 class="pb-0">
        <h4>Ingrédients</h4>
      </v-col>
      <v-col cols=12 class="pt-0">
        <v-list>
          <v-list-item>
            <v-list-item-content>
              Pâtes: 200 gr
            </v-list-item-content>
          </v-list-item>
          <v-list-item>
            <v-list-item-content>
              Sauce tomate: 200 gr
            </v-list-item-content>
          </v-list-item>
          <v-list-item>
            <v-list-item-content>
              Steak haché: 100 gr
            </v-list-item-content>
          </v-list-item>
          <v-list-item>
            <v-list-item-content>
              Huile d'olives: 3 cac
            </v-list-item-content>
          </v-list-item>
        </v-list>
      </v-col>
    </v-row>
  </v-container>
</template>

<script>
import Calories from '@/components/icons/Calories.vue';
import Timer from '@/components/icons/Timer.vue';

export default {
  name: 'recipe-details',
  components: {
    Calories,
    Timer,
  },
};
</script>
