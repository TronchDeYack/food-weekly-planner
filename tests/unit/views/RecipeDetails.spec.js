import { shallowMount } from '@vue/test-utils';
import { when } from 'jest-when';

import RecipeDetails from '@/views/RecipeDetails.vue';
import { getRecipe } from '@/api';

jest.mock('@/api');

describe('RecipeDetails', () => {
  const recipeId = 'ID-DU-Q';
  const recipe = {
    id: recipeId,
    title: 'Spaghetti bolognaise',
    minutes: 20,
    caloriesLevel: 'medium',
    ingredients: [
      {
        id: 'PATE-ID',
        label: 'Pâtes',
        quantity: 200,
        unit: 'gr',
        type: 'starchy-food',
      },
      {
        id: 'SAUCE-TOMATE-ID',
        label: 'Sauce Tomate',
        quantity: 200,
        unit: 'gr',
        type: 'condiment',
      },
      {
        id: 'STEAK-ID',
        label: 'Steak haché',
        quantity: 100,
        unit: 'gr',
        type: 'meat',
      },
      {
        id: 'OIL-ID',
        label: 'Huile d\'olives',
        quantity: 3,
        unit: 'cac',
        type: 'condiment',
      },
    ],
  };

  const propsData = {
    id: recipeId,
  };

  let component;

  beforeEach(() => {
    when(getRecipe).calledWith(recipeId).mockResolvedValue(recipe);
    component = shallowMount(RecipeDetails, { propsData });
  });

  it('Should render.', () => {
    expect(component).toMatchSnapshot();
  });

  describe('State', () => {
    it('Should initialize "recipe" state to null.', () => {
      const cmp = shallowMount(RecipeDetails, { propsData });
      expect(cmp.vm.$data.recipe).toBeNull();
    });
  });

  describe('mounted', () => {
    it('Should get recipe data from API and store them in "recipe" state.', () => {
      expect(component.vm.$data.recipe).toEqual(recipe);
    });
  });
});
