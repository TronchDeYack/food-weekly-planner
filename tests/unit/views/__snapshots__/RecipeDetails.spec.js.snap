// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipeDetails Should render. 1`] = `
<v-container-stub tag="div">
  <v-row-stub tag="div">
    <v-col-stub cols="12" tag="div">
      <h3>Spaghetti Bolognaise</h3>
    </v-col-stub>
  </v-row-stub>
  <v-row-stub tag="div" align="center">
    <v-col-stub tag="div" class="d-flex align-center">
      <timer-stub></timer-stub>
      20 min
    </v-col-stub>
    <v-col-stub tag="div">
      <calories-stub class="medium"></calories-stub>
    </v-col-stub>
  </v-row-stub>
  <v-divider-stub></v-divider-stub>
  <v-row-stub tag="div">
    <v-col-stub cols="12" tag="div" class="pb-0">
      <h4>Ingrédients</h4>
    </v-col-stub>
    <v-col-stub cols="12" tag="div" class="pt-0">
      <v-list-stub tag="div" tile="true">
        <v-list-item-stub activeclass="" tag="div">
          <v-list-item-content-stub>
            Pâtes: 200 gr
          </v-list-item-content-stub>
        </v-list-item-stub>
        <v-list-item-stub activeclass="" tag="div">
          <v-list-item-content-stub>
            Sauce tomate: 200 gr
          </v-list-item-content-stub>
        </v-list-item-stub>
        <v-list-item-stub activeclass="" tag="div">
          <v-list-item-content-stub>
            Steak haché: 100 gr
          </v-list-item-content-stub>
        </v-list-item-stub>
        <v-list-item-stub activeclass="" tag="div">
          <v-list-item-content-stub>
            Huile d'olives: 3 cac
          </v-list-item-content-stub>
        </v-list-item-stub>
      </v-list-stub>
    </v-col-stub>
  </v-row-stub>
</v-container-stub>
`;
